packages/venia-ui/lib/util/__tests__/productVariants.spec.js

```diff
@@ -0,0 +1,66 @@
+import {
+    findMatchingProductOption,
+    findMatchingProductOptionValue
+} from '../productVariants';
+
+import { product, variant } from '../__fixtures__/product';
+
+describe('findMatchingProductOption', () => {
+    const findMatch = findMatchingProductOption;
+
+    test('it finds matches successfully', () => {
+        const variantOption = variant.options[0];
+
+        const result = findMatch({ product, variantOption });
+
+        expect(result).toBeInstanceOf(Object);
+    });
+
+    test('it returns undefined when no match is found', () => {
+        const variantOption = Object.assign({}, variant.options[0], {
+            label: 'NO_MATCH '
+        });
+
+        const result = findMatch({ product, variantOption });
+
+        expect(result).toBeUndefined();
+    });
+});
+
+describe('findMatchingProductOptionValue', () => {
+    const findMatch = findMatchingProductOptionValue;
+
+    test('it finds matches successfully', () => {
+        const variantOption = variant.options[0];
+
+        const result = findMatch({ product, variantOption });
+
+        expect(result).toBeInstanceOf(Object);
+        expect(result.option).toBeDefined();
+        expect(result.value).toBeDefined();
+    });
+
+    test('it returns the correct object when no match is found due to no matching option', () => {
+        const variantOption = Object.assign({}, variant.options[0], {
+            label: 'NO_MATCH '
+        });
+
+        const result = findMatch({ product, variantOption });
+
+        expect(result).toBeInstanceOf(Object);
+        expect(result.option).toBeUndefined();
+        expect(result.value).toBeUndefined();
```

Comment:
If no matching option can be found, it would be better API design to have this match `findMatchingProductOption` and return `undefined` instead of an object with two properties that are `undefined`.